test/extended/builds/pipeline.go

```diff
@@ -177,6 +178,55 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
 		})
 	})
 
+	g.Context("Pipeline with env vars", func() {
+		g.AfterEach(func() {
+			if os.Getenv(jenkins.DisableJenkinsGCStats) == "" {
+				g.By("stopping jenkins gc tracking")
+				ticker.Stop()
+			}
+		})
+
+		g.It("should build and complete successfully", func() {
+			// instantiate the bc
+			g.By(fmt.Sprintf("calling oc new-app -f %q", envVarsPipelinePath))
+			err := oc.Run("new-app").Args("-f", envVarsPipelinePath).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// start the build
+			g.By("starting the pipeline build and waiting for it to complete")
+			br, _ := exutil.StartBuildAndWait(oc, "sample-pipeline-withenvs")
+			debugAnyJenkinsFailure(br, oc.Namespace()+"-sample-pipeline-withenvs", oc, true)
+			br.AssertSuccess()
+
+			g.By("get build console logs and see if succeeded")
+			_, err = j.WaitForContent("Finished: SUCCESS", 200, 10*time.Minute, "job/%s-sample-pipeline-withenvs/lastBuild/consoleText", oc.Namespace())
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("get build console logs and see if env is set")
+			_, err = j.WaitForContent("FOO1 is BAR1", 200, 10*time.Minute, "job/%s-sample-pipeline-withenvs/lastBuild/consoleText", oc.Namespace())
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("get build console logs and see if env is still not set")
+			_, err = j.WaitForContent("FOO2 is null", 200, 10*time.Minute, "job/%s-sample-pipeline-withenvs/lastBuild/consoleText", oc.Namespace())
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// start the next build
+			g.By("starting the pipeline build, including env var, and waiting for it to complete")
+			br, _ = exutil.StartBuildAndWait(oc, "-e", "FOO2=BAR2", "sample-pipeline-withenvs")
+			debugAnyJenkinsFailure(br, oc.Namespace()+"-sample-pipeline-withenvs", oc, true)
+			br.AssertSuccess()
+
+			g.By("get build console logs and see if succeeded")
+			_, err = j.WaitForContent("Finished: SUCCESS", 200, 10*time.Minute, "job/%s-sample-pipeline-withenvs/lastBuild/consoleText", oc.Namespace())
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("get build console logs and see if env is set")
+			_, err = j.WaitForContent("FOO2 is BAR2", 200, 10*time.Minute, "job/%s-sample-pipeline-withenvs/lastBuild/consoleText", oc.Namespace())
+			o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
should run a third build w/ no env args and confirm that FOO2 is back to null(? emptystring?).  (or just reverse the order of the existing 2 tests should be sufficient).